File: leetcode/0150-evaluate-reverse-polish-notation/solution.java
import java.util.ArrayDeque;
import java.util.Deque;

class Solution {
    public int evalRPN(String[] tokens) {
        
        Deque<Integer> st = new ArrayDeque<>();
        
        for(String t : tokens){
            
            switch(t){
                    
                case "+":
                    
                    st.push(st.pop()+st.pop());
                    break;
                    
                case "-":{
                    
                    int a = st.pop();
                    int b = st.pop();
                    
                    st.push(b-a);
                    break;
                }    
                case "*":
                    
                    st.push(st.pop()*st.pop());
                    break;
                    
                case "/":{
                    
                    int a = st.pop();
                    int b = st.pop();
                    
                    st.push(b/a);
                    break;
                }
                default:
                    
                    st.push(Integer.parseInt(t));
            }
        }
        return st.pop();
    }
}

File: leetcode/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        s =[]
        
        for token in tokens:    

            if token not in ('+','-','*','/'):

                s.append(token)

            else:

                A = s.pop()
                B = s.pop()
                a = int(A)
                b = int(B)

                if token == '+':

                    t = a + b

                    s.append(t)

                elif token == '-':

                    t = b - a 

                    s.append(t)

                elif token == '/':

                    t = b / a
                    s.append(t)

                elif token == '*':

                    t = a * b
                    s.append(t)

        ans = s.pop()
        return int(ans)

                    


